src/Arcus.BackgroundJobs.AzureActiveDirectory/ClientSecretExpirationInfoProvider.cs

```diff
@@ -72,13 +66,22 @@ public async Task<List<ApplicationWithExpiredAndAboutToExpireSecrets>> GetApplic
                         }
                     }
                 }
-                else
-                {
-                    _logger.LogTrace("Application {ApplicationName} has no secrets", application.DisplayName);
-                }
             }
 
             return applicationsList;
         }
+
+        private static double DetermineRemainingDaysBeforeExpiration(PasswordCredential password)
+        {
+            DateTimeOffset? expirationDate = password.EndDateTime;
+            if (expirationDate is null)
+            {
+                return 0;
```

Comment:
When no expiration date is present, should we consider it being expired...

Reply:
I don't think we should, if I remember correctly it used to be possible to set secrets to be valid forever. Found this qoute here https://docs.microsoft.com/en-us/answers/questions/422538/future-plans-of-microsoft-with-the-maximum-expirat.html: 

> The "Never Expires" option has been replaced by "Custom " option. This was done because Microsoft was running into a problem where badly provisioned or abandoned configurations were exposing parts of the Azure AD platform and this was a way for them to mitigate abandonment. It was difficult to achieve this with the original "never" configured secrets.

In that case the expiration date might not be present, we would mark it as expired but in reality it still is valid.

Reply:
Aha, ok, then we should change this, indeed, my bad. 😄 Thanks for noting.

Reply:
Updated this so we skip secrets without expiration date.